software/displayerang.livecodescript

```diff
@@ -0,0 +1,1402 @@
+         
+      end if
+   end repeat
+end tempConvertData
+
+on middleOut pData,@pDraw,@pFocus -- ! a hacky way to draw the provided graph, calls assignLayers
+   -- ! as of 10JAN2016 middleOut only follows flow links, so it won't visit or follow dive/rise links
+   put "on middleOut" into xX["write"];    devLog xX
+   -- pData ! just sGraphData at the moment, maybe this should be an adjacency list ?
+   -- setup variables --------------------------------------------------------------------------------
+   put empty into pData["layers"]
+   put empty into pData["visited"]
+   -- make sure the focus node is actually focusable
+   put the keys of pData["nodes"] into tKeys
+   put empty into tRecs
+   repeat for each line tLine in tKeys
+      if tLine is a number then put tLine into line (the number of lines of tRecs)+1 of tRecs
+   end repeat
+   if pFocus is not among the lines of tRecs then
+      sort tRecs numeric descending
+      put line 1 of tRecs into pFocus
+   end if
+   -- recursively assign layers ------------------------------------------------------------------------
+   -- pass pData by reference, and focus node, and initial layer
+   -- add layer and visited info to pData
+   assignLayers pData,pFocus,0
+   -- remove duplicates in layers ------------------------------------------------------------------
+   repeat for each key tLayer in pData["layers"]
+      put pData["layers"][tLayer]into tOneLayer
+      split tOneLayer by comma
+      combine tOneLayer using comma
+      if char -1 of tOneLayer is not "," then put "," after tOneLayer
+      put tOneLayer into pData["layers"][tLayer]
+   end repeat
+   -- shuffle nodes so links point left-to-right ----------------------------------------------------
+   -- when a pair of linked nodes are misaligned...
+   --- if on the pos side (layers >0), shift end more pos
+   --- if on the neg side (layers <0), shift start more neg
+   --- making a link vertical is okay, but not backwards
+   --- don't move the focused node out of layer 0
+   put empty into tShuffles -- cr delimited list; node and direction shuffled; kill infinite loop
+   put true into tAgain -- keep repeating until nothing changes
+   repeat while tAgain is true
+      put false into tAgain
+      repeat for each item tLink in pData["visited"]["links"]
+         put item 1 of pData["links"][tLink]["prev"] into tStart
+         put item 1 of pData["links"][tLink]["next"] into tEnd
+         put pData["nodes"][tStart]["layer"] into tStartLayer
+         put pData["nodes"][tEnd]["layer"] into tEndLayer
+         if not (tStartLayer < tEndLayer) then
+            if tEnd is pFocus then
+               -- shuffle start more negative
+               put true into tAgain
+               --               delete item itemOffset(tStart,pData["layers"][tStartLayer]) of pData["layers"][tStartLayer]
+               set the wholeMatches to true
```

Comment:
@matthewmaier You are setting `wholeMatches` here but on the next repeat loop it will impact the itemOffset called in the other conditions. If you want it true for all conditions move it outside the loop otherwise reset it to false after itemOffset.